### PR TITLE
feat(docker): Support `releaseTimestamp` for DockerHub tag results

### LINKED ISSUE #24135: Include digest and releaseTimestamp from Docker Hub result
### Describe the proposed change(s).

Fetch not only the tag name but also its digest and release timestamp when using the hub.docker.com tags API

```
export interface DockerHubTags {
  next?: string;
  results: {
    name: string;
    digest: string;
    tag_last_pushed: string;
  }[];
}
```

It could be a good time to use `zod` too.

### Describe why we need/want these change(s).

Fetching digest will improve efficiency. Fetching timestamp will allow us to use minimumReleaseAge

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Closes: #24135

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
I saw:

```
DEBUG: Docker: Failed to parse some tags from Docker Hub (repository=renovate-reproductions/dockertags1)
       "error": {
         "message": "Schema error",
         "stack": "ZodError: Schema error\n    at Object.transform (/Users/rhys/src/github/renovatebot/renovate/lib/util/schema-utils.ts:65:21)\n    at /Users/rhys/src/github/renovatebot/renovate/node_modules/.pnpm/zod@3.22.2/node_modules/zod/lib/types.js:2870:51\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
         "issues": {
           "492": {"tag_last_pushed": "Expected string, received null"},
           "493": {"tag_last_pushed": "Expected string, received null"},
           "494": {"tag_last_pushed": "Expected string, received null"},
           "___": "... 6 more"
         }
       }
```

I think it's best to silently ignore this missing field

File: lib/modules/datasource/docker/schema.ts [hunk 1]
```diff
@@ -127,3 +130,35 @@ export const Manifest = z.union([
 ]);
 
 export type Manifest = z.infer<typeof Manifest>;
+
+export const DockerHubTag = z
+  .object({
+    name: z.string(),
+    tag_last_pushed: z.string().nullable().catch(null),
```
Comment (viceice) [inline lib/modules/datasource/docker/schema.ts:137]:
```suggestion
    tag_last_pushed: z.string().datetime().nullable().catch(null),
```
should be a valid iso timestamp

**Review by viceice — COMMENTED** [review]
what about the digest?

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Digest is more complicated, I'm ok to do that separately

**Pull request merged**